notebooks/ReFT_Adapters_Finetuning.ipynb

```diff
@@ -0,0 +1,505 @@
+       "      [24544/24544 3:40:48, Epoch 2/2]\n",
+       "    </div>\n",
+       "    <table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       " <tr style=\"text-align: left;\">\n",
+       "      <th>Epoch</th>\n",
+       "      <th>Training Loss</th>\n",
+       "      <th>Validation Loss</th>\n",
+       "      <th>Accuracy</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <td>1</td>\n",
+       "      <td>0.603000</td>\n",
+       "      <td>0.508186</td>\n",
+       "      <td>0.795619</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <td>2</td>\n",
+       "      <td>0.559800</td>\n",
+       "      <td>0.475799</td>\n",
+       "      <td>0.812124</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table><p>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/plain": [
+       "TrainOutput(global_step=24544, training_loss=0.6184384013092658, metrics={'train_runtime': 13249.0278, 'train_samples_per_second': 59.28, 'train_steps_per_second': 1.853, 'total_flos': 2.0971423089834394e+17, 'train_loss': 0.6184384013092658, 'epoch': 2.0})"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "trainer = AdapterTrainer(\n",
+    "    model=model,\n",
+    "    args=training_args,\n",
+    "    data_collator=data_collator,\n",
+    "    train_dataset=train_dataset,\n",
+    "    eval_dataset=eval_dataset,\n",
+    "    tokenizer=tokenizer,\n",
+    "    compute_metrics = compute_metrics\n",
+    ")\n",
+    "\n",
+    "trainer.train()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With only as little as  0.030% of the parameters, we were able to successfully achieve 0.81 accuracy within only two epochs! We also reduced the time to train the model drastically than if we were to fully fine-tune the model using traditional methods."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Inference\n",
+    "\n",
+    "With our tuned model using `reft` adapters, we can now do some inference on some new text!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "mapping = {\n",
+    "    0: \"neutral\",\n",
+    "    1: \"entailment\",\n",
+    "    2: \"contradiction\"\n",
+    "}\n",
+    "\n",
+    "def infer_text(text):\n",
+    "    input_ids = tokenizer(text, truncation=True, padding='max_length')\n",
+    "    input_ids[\"input_ids\"] = torch.tensor(input_ids[\"input_ids\"])\n",
+    "    input_ids[\"attention_mask\"] = torch.tensor(input_ids[\"attention_mask\"])\n",
```

Comment:
can add `return_tensors="pt"` (same as in preprocess_function) to tokenizer call so you don't need to convert to tensors afterwards

Reply:
ah, good catch!